x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_creation/engine_creation.tsx

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+
+import {
+  EuiForm,
+  EuiFlexGroup,
+  EuiFormRow,
+  EuiFlexItem,
+  EuiFieldText,
+  EuiSelect,
+  EuiPageHeader,
+  EuiPageHeaderSection,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+  EuiButton,
+  EuiPanel,
+} from '@elastic/eui';
+import { useActions, useValues } from 'kea';
+
+import { SetAppSearchChrome as SetPageChrome } from '../../../shared/kibana_chrome';
+import { FlashMessages } from '../../../shared/flash_messages';
+import {
+  ALLOWED_CHARS_NOTE,
+  ENGINE_CREATION_FORM_ENGINE_LANGUAGE_LABEL,
+  ENGINE_CREATION_FORM_ENGINE_NAME_LABEL,
+  ENGINE_CREATION_FORM_ENGINE_NAME_PLACEHOLDER,
+  ENGINE_CREATION_FORM_SUBMIT_BUTTON_LABEL,
+  ENGINE_CREATION_FORM_TITLE,
+  ENGINE_CREATION_TITLE,
+  SANITIZED_NAME_NOTE,
+  SUPPORTED_LANGUAGES,
+} from './constants';
+import { EngineCreationLogic } from './engine_creation_logic';
+
+export const EngineCreation: React.FC = () => {
+  const { name, rawName, language } = useValues(EngineCreationLogic);
+  const { setLanguage, setRawName, submitEngine } = useActions(EngineCreationLogic);
+
+  return (
+    <div data-test-subj="EngineCreation">
```

Comment:
[not a change request, just thinking out loud] I'm kinda torn on these just because I wish they didn't create an extra unnecessary div wrapper. Can we consider putting this `data-test-subj="EngineCreation"` on say, the `EuiPageHeader` or `EuiPageBody`? We're just using them as a very basic "page loaded" landmark check in our tests (both Jest & Cypress), right?

Reply:
I think it makes sense at the view level labelling the entire container, I don't think the extra div matters.